### PR TITLE
Test cliff_gap and cliff_evaluated varaibles

### LINKED ISSUE #3029: Incorrect cliff_gap calculation
The `cliff_gap` variable is untested in the 0.469.0 version of the model.  When adding the following test, the calculated `cliff_gap` is over $78,000 instead of the expected value of zero.

```
- name: household_net_income test TX-netinc-base
  absolute_error_margin: 0.01
  period: 2021
  input:
    age: 40
    employment_income: 100_000
    taxable_interest_income: 100_000
    state_fips: 48  # TX
  output:
    spm_unit_net_income: 151_539.00
    household_net_income: 151_539.00

- name: household_net_income test TX-netinc-alt
  absolute_error_margin: 0.01
  period: 2021
  input:
    age: 40
    employment_income: 102_000
    taxable_interest_income: 100_000
    state_fips: 48  # TX
  output:
    spm_unit_net_income: 152_670.00
    household_net_income: 152_670.00

- name: cliff_gap test TX-cliff_gap
  absolute_error_margin: 0.01
  period: 2021
  input:
    age: 40
    employment_income: 100_000
    taxable_interest_income: 100_000
    state_fips: 48  # TX
  output:  # as expected from above two test results
    cliff_gap: 0
    # using PEUS 0.469.0, cliff_gap test results are:
    #
    # Test 'cliff_gap test TX-cliff_gap':
    #   cliff_gap@2021: [78752.61] differs from 0.0
    #                   with an absolute margin [78752.61] > 0.01
```

Note that there is also a bug in the formula of the untested `cliff_evaluated` variable.

### LINKED ISSUE #3134: Test the cliff_gap and cliff_evaluated variables
There is no testing of the `cliff_gap` or `cliff_evaluated` variables.

### PR DESCRIPTION
**Pull request by martinholmer**
Fixes #3134.

### CODE REVIEW

File: policyengine_us/variables/household/cliff.py [hunk 1]
```diff
@@ -6,50 +6,53 @@ class cliff_evaluated(Variable):
...
-    documentation = "Amount of income lost if this person's employment income increased by $2,000."
+    documentation = "Amount of income lost if this person's employment income increased by delta amount."
     definition_period = YEAR
 
     def formula(person, period, parameters):
-        simulation = person.simulation
-        alt_simulation = simulation.get_branch(f"all_adults_2k_pay_rise")
-        delta = parameters(period).simulation.cliff_threshold
-        for variable in simulation.tax_benefit_system.variables:
-            if variable not in simulation.input_variables:
```
[the remark this answers is not in the document]
Comment (martinholmer) [reply policyengine_us/variables/household/cliff.py:31]: The "justification" for the switch to `is_input_variable()`, in both this PR and in the MTR fix PR #3036, is that this change makes the failing tests (see issues #3017 and #3029) pass.  The problem with the `input_variables()` method is  that it always returns an empty list, which causes the wildly inaccurate results reported in issues #3017 and #3029.

**Review by nikhilwoodruff — APPROVED** [review]
LGTM, thanks Martin.
